static/js/compose_actions.js

```diff
@@ -416,21 +416,19 @@ exports.quote_and_reply = function (opts) {
         // are prone to glitches where you select the
         // text, plus it's a complicated codepath that
         // can have other unintended consequences.)
-        //
-        // Note also that we always put the quoted text
-        // above the current text, which explains us
-        // moving the caret below.  I think this is what
-        // most users will want, and it's consistent with
-        // the behavior we had on FF before this change
-        // (which may have been an accident of
-        // implementation).  If we change this decision,
-        // we'll need to make `insert_syntax_and_focus`
-        // smarter about newlines.
-        textarea.caret(0);
+
+        if (textarea.caret() !== 0) {
+            // Insert a newline before quoted message if there is
+            // already some content in the compose box and quoted
+            // message is not being inserted at the beginning.
+            textarea.caret("\n");
+        }
     } else {
         exports.respond_to_message(opts);
     }
 
+    const prev_caret = textarea.caret();
```

Comment:
Should we create this variable above the last block, so that we can use it?  Or is the intent that the value be after the newline that is potentially added above?

Reply:
Yes, the intent here is to take the value after the potential newline from the previous block is added. If we put this before the previous block, then we'll have to modify `prev_caret` if the newline is added.